pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/BaseTransformFunction.java

```diff
@@ -226,6 +246,27 @@ public double[] transformToDoubleValuesSV(ProjectionBlock projectionBlock) {
     return _doubleValuesSV;
   }
 
+  @Override
+  public BigDecimal[] transformToBigDecimalValuesSV(ProjectionBlock projectionBlock) {
+    int length = projectionBlock.getNumDocs();
+    if (_bigDecimalValuesSV == null || _bigDecimalValuesSV.length < length) {
+      _bigDecimalValuesSV = new BigDecimal[length];
+    }
+
+    Dictionary dictionary = getDictionary();
+    if (dictionary != null) {
+      int[] dictIds = transformToDictIdsSV(projectionBlock);
+      dictionary.readBigDecimalValues(dictIds, length, _bigDecimalValuesSV);
+    } else {
+      if (getResultMetadata().getDataType().getStoredType() == DataType.STRING) {
```

Comment:
Should we also support reading big decimal from other numeric types?

Reply:
yes, I think so. I had it in my original PR. Will add back.